From a249e3c1cc402b7b6b674ad0fd89613189e1c2db Mon Sep 17 00:00:00 2001
From: Luca Boccassi <bluca@debian.org>
Date: Sat, 15 Dec 2018 10:23:37 +0000
Subject: [PATCH 1/4] Makefile: have check target depend on all

Otherwise it will simply fail immediately from a just-cleaned
workspace:

$ make check -j1
cd testsuite && make && make alltests
echo "Entering iproute2" && cd iproute2 && make configure && cd ..;
Entering iproute2
make -C tools
Makefile:3: ../../config.mk: No such file or directory
make[2]: *** No rule to make target '../../config.mk'.  Stop.

Fixes: 8804a8c0d387 ("Makefile: Add check target")

Signed-off-by: Luca Boccassi <bluca@debian.org>
Reviewed-by: Petr Vorel <petr.vorel@gmail.com>
Tested-by: Petr Vorel <petr.vorel@gmail.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index b7488add..20c760e2 100644
--- a/Makefile
+++ b/Makefile
@@ -119,7 +119,7 @@ clobber:
 
 distclean: clobber
 
-check:
+check: all
 	cd testsuite && $(MAKE) && $(MAKE) alltests
 
 cscope:
-- 
2.19.2

